sphinx/builders/linkcheck.py

```diff
@@ -230,14 +234,21 @@ def check():
             if uri is None:
                 break
             status, info, code = check()
-            self.rqueue.put((uri, docname, lineno, status, info, code))
+            self.rqueue.put(
+                OrderedDict((
+                    ('status', status), ('code', code), ('uri', uri),
+                    ('info', info), ('docname', docname), ('lineno', lineno))
+                ))
```

Comment:
No reason to change the data structure of `self.rqueue`. Could you pass these variables to `write_json_entry()` instead of `result` please?

Reply:
I've replaced the one dict with the original immutable tuple and an additional dict of refs.